### DISCUSSION TITLE
Relay email from on prem exchange server

### DISCUSSION THREAD

**Discussion by jpcapone** [postalserver/postal#2455] [Help with using Postal]
My ISP blocks outbound port 25.  I would like to use port 2525 to relay email from my on prem exchange server to my VPS postal server.  I would like inbouind port 25 email that hits my Postal VPS to be forwarded to my on prem exchange server on a non stand port.  I am struggling with figuring out how to make this happen.  Any suggestions would be appreciated.

**Comment by jpcapone** [comment]
Thus far i am able to send test messages from my Postal instance.  I can also use external smtp test to send through the Postal service.  I think i am just missing some fundamental configuration pieces.  I am seeing indications that  an entry needs to be made in the postal.yml.  Also when i reply to test messages i get 
`530 Authentication required`

I am gonna keep working through this but any assistance would be appreciated.

**Comment by willpower232** [comment]
If your ISP is blocking the port then you may be struggling in general. How does your on prem exchange server send messages if the ISP is blocking it?

I would imagine you're seeing that message because Postal can't tell the difference between a message you're trying to relay and a message you're trying to send as Postal was originally intended for.

**Reply by jpcapone** [reply]
Thanks for getting back to me.  I have everything working to an extent.  I implemented port fowarding from 2525 to port 25, on the Postal server, and I can send outbound emails from Exchange to Postal on port 2525 and receive inbound messages in Exchange from Postal on port 2525.  Outbound from Exchange is flawless while I am using smoke and mirrors to get email inbound from Postal.   When I say smoke and mirrors here is what I mean:

In Postal under Routing I have email addresses for on prem user accounts and I made the endpoint the DNS name of my Exchange server listening on port 2525 and obviously I have a receive connector configured and scoped to receive these messages.  I think this negates the `530 Authentication required`as the messages are recieved

In my postal.yml file I have this section referring to Exchange on prem:
```
`smtp:
      host: exchange.XXX.com
      Port: 2525
      username: # Complete when Postal is running and you can
      password: # generate the credentials within the interface.
      from_name: Postal
      from_address: XXX.XXX.com
`
```
I don't have any sections referrencing `smtp_relay`

So how does this sound to you?  I would like to NOT have to use Routes for inbound messages to Exchange on prem but I can't think of a better way to do it.  I would like to be able to send from Exchange and have the replies come back to the sender without having to have a specific route. 

EDIT;  I confirmed that I can wildcard route email to the on prem server on port 2525.  I would still like to know if my config makes sense.  Should I be using `smtp_relay:` instead of `smtp:`?

**Reply by willpower232** [reply]
The `smtp` section is for Postal sending its own emails (i.e. password resets)

If you need to relay messages out of Postal to get to the internet then you can use `smtp_relay` which looks like this https://github.com/postalserver/postal/blob/8b525d0381a9e0113af808b9ec2eb47bf78ec60b/config/postal.defaults.yml#L79-L83

**Reply by jpcapone** [reply]
Thanks for getting back to me.  I currently have Postal listening on port 25 and port 2525 and this does give me the ability to relay to an Exchange server that is stuck behind and IP where port 25 is blocked.  in my config the only reference to my on prem server is under `smtp:` and it looks like this
```
smtp:
  host: OnPrem.ExchangeServer.com
  port: 2525
  username: # Complete when Postal is running and you can
  password: # generate the credentials within the interface.
  from_name: Postal
  from_address: my.postalserver.com  
```

Considering that this config is working I am hesistant to make a change however if you think i should swap out the config above and attempt to use the `smtp_relays:` stanza and test it I would do that.  Let me know
